pkg/resources/task.go

```diff
@@ -0,0 +1,478 @@
+package resources
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/csv"
+	"fmt"
+	"strings"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/pkg/errors"
+)
+
+const (
+	taskIDDelimiter = '|'
+)
+
+var taskSchema = map[string]*schema.Schema{
+	"name": &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Specifies the identifier for the task; must be unique for the database and schema in which the task is created.",
+		ForceNew:    true,
+	},
+	"database": &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The database in which to create the task.",
+		ForceNew:    true,
+	},
+	"schema": &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The schema in which to create the task.",
+		ForceNew:    true,
+	},
+	"warehouse": &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The warehouse the task will use.",
+		ForceNew:    false,
+	},
+	"schedule": &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "The schedule for periodically running the task. This can be a cron or interval in minutes.",
+	},
+	"session_parameters": &schema.Schema{
+		Type:        schema.TypeSet,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Specifies a comma-separated list of session parameters to set for the session when the task runs. A task supports all session parameters.",
```

Comment:
It is not clear to me how this is supposed to be used?  A set of strings, each of which has a comma-separated list of paremeters?

Would `schema.TypeMap` be clearer?

Reply:
Yes that would make more sense! It would simplify the logic of removing session parameters as well! Thanks, didn't think of `schema.TypeMap`